### PR TITLE
BHoM managed Python virtualenvs

### PR DESCRIPTION
**Pull request by tg359**
### NOTE: Depends on 
https://github.com/BHoM/Python_Toolkit/pull/85
   
### Issues addressed by this PR

Closes #

Handling of Python environments via BHoM rationalised around a base "full" environment, and virtualenvs for any toolkit/project-specific environment that can be pre-defined with custom codebases via a referenced local-package added upon installation.

This PR creates a virtualenv with preinstalled commonly used Python ML packages. It also removes a lot of older code that is unused in an attempt to clean-up this toolkit ready for any newcomers to start adding aligned with the new Python management framework.

### Test files

[83.zip](https://github.com/BHoM/LadybugTools_Toolkit/files/9358146/83.zip)

### CODE REVIEW

File: MachineLearning_Engine/Compute/MachineLearningToolkitPythonEnvironment.cs [hunk 1]
```diff
@@ -20,28 +20,31 @@
...
-        [Output("Object", "The probabilities (score) for each class. Use the Query.ImageClasses() method to match ")]
-        public static string RecogniseObject(string imagePath, bool gpu = false)
+        [Description("MachineLearning_Toolkit\nMethod used to create the Python environment used to run all Python scripts within this toolkit.")]
+        [Input("run", "Starts the installation of the toolkit if true. Stays idle otherwise.")]
+        [Output("pythonEnvironment", "The MachineLearning_Toolkit Python environment.")]
+        public static PythonEnvironment MachineLearningToolkitPythonEnvironment(bool run = false)
         {
-            return BH.Engine.MachineLearning.Base.Compute.Invoke(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.Namespace, "RecogniseObject.infer", imagePath, gpu).As<string>();
+            if (run)
+            {
```
Comment (alelom) [inline MachineLearning_Engine/Compute/MachineLearningToolkitPythonEnvironment.cs]: Please remove indented block. Perhaps consider adding a warning to alert user to toggle `run` to `true`.
```suggestion
            if (!run)
            {
               BH.Engine.Base.Compute.RecordWarning($"Please toggle `{nameof(run)}` to true.");
               return null;
            }
```

**Review by alelom — CHANGES_REQUESTED** [review]
Looks great, still need to test it, minor changes requested

File: MachineLearning_Engine/Compute/MachineLearningToolkitPythonEnvironment.cs (hunk 1 above)
Comment (tg359) [reply MachineLearning_Engine/Compute/MachineLearningToolkitPythonEnvironment.cs]: Just realised that changes in the Python_Tk should mean that we don't have to `if(run)` check any of the dependant methods.

**Review by alelom — DISMISSED** [review]
Thanks for the code changes, code looks good to me.
I am approving on the basis of code changes and on the provided test file, see considerations done in https://github.com/BHoM/Python_Toolkit/pull/85#pullrequestreview-1085407145, given that the test file is the same both PRs.

**Comment by tg359** [comment]
@BHoMBot check required

File: MachineLearning_Engine/Compute/InstallPythonML.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public static partial class Compute
         [Description("MachineLearning_Toolkit\nMethod used to create the Python environment used to run all Python scripts within this toolkit.")]
         [Input("run", "Starts the installation of the toolkit if true. Stays idle otherwise.")]
         [Output("pythonEnvironment", "The MachineLearning_Toolkit Python environment.")]
-        public static PythonEnvironment MachineLearningToolkitPythonEnvironment(bool run = false)
+        public static PythonEnvironment InstallPythonML(bool run = false)
```
Comment (alelom) [inline MachineLearning_Engine/Compute/InstallPythonML.cs]: Didn't want to include the word `Env` in it? Thought it was better to have it - we're not simply installing a Python distribution, we are actually creating an environment for it no?

Just wondering as the other day we said we'd go for `InstallPyEnv_ML`, `InstallPyEnv_LBT`, etc.

Comment (alelom) [reply MachineLearning_Engine/Compute/InstallPythonML.cs]: I actually like the fact you included the entire word "Python" in the name (easier to search), but I think we should mention "Environment" or "Env" for clarity? 

Maybe `InstallPythonEnv_ML`, or `InstallPythonEnvironment_ML`?

**Comment by tg359** [comment]
@BHoMBot check compliance

**Comment by tg359** [comment]
@BHoMBot check compliance

**Comment by tg359** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Review by FraserGreenroyd — APPROVED** [review]
Approving based on @alelom review as commits since then have been versioning and compliance.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**